azbrscca/azbrscca.org
# Moving all MTC things over to AZBR

## Why are we doing this?

We're transferring the registration from MTC site back to AZBR. This PR covers code changes to the AZBR site to reflect that the new URL will be `http://registration.azbrscca.org/`. It also removes references to the MTC name.

~This is in draft status and not ready to merge yet.~

## How can someone view these changes?

The [dev site](http://dev.azbrscca.org/) is currently using this branch. I manually updated the `Common.php` file on the server as that is not in version control.

Steps to manually verify the change:

The results page is a good way to verify that this is hitting `http://registration.azbrscca.org/`:
1. Open the Chrome developer console (`command` + `option` + `j` on OSX)
2. Go to the _Network_ tab
3. Go to the _Results_ page on the dev site: http://dev.azbrscca.org/autocross/results.html
4. Near the bottom of the list of requests, there will be one in the format `<eventId>?order_by=pax_time` such as `422?order_by=pax_time`. Double click this request so that is opens in another browser window.
5. The url should start with `http://dev.azbrscca.org/reg-api/` and the content should be JSON data of the event results. Previously, this would be a URL going to `mindthecones.com`

Please bounce around the site and see if anything looks out of sorts. The _Results_ page is the heaviest hitter of the API & data. The autocross and rallycross calendars and the front page are the only other culprits (we could toggle those to 2020?. You can also navigate around [the new registration site](http://registration.azbrscca.org/) just note it is hooked up to **live** data.

## What possible risks or adverse effects are there?

Nothing major. If I forgot a reference somewhere with this PR, it can be corrected without any data issues. AZBR treats the registration site as read-only. Someone has to go to the registration site and log in to change data. Also, right now, both MTC and `registration.azbrscca.org` are pointed to the same database on AZBR's 1and1 account so that no data gets out of sync.

## What are the follow-up tasks?

Manual testing is the main task. There are a few things I need to complete for the migration of MTC back to AZBR, but those impact the registration site, nothing with the AZBR site and this PR.

When this is ready to go live:
`$HOME/config/Common.php.live` requires an update similar to `$HOME/config/Common.php.dev`. 

`$HOME/config/Common.php.live` and the `.htaccess.live` file in the `reg-api` subdirectory will be copied to their correct locations when the `configure` script runs. The usual deployment process to the [live site](https://github.com/azbrscca/azbrscca.org/wiki/Webmaster-Notes#deploying-to-the-live-site) will handle this.

## Are there any known issues?

~None so far. I want to run through the site more to see what problems I find. I'll remove the `DRAFT` status from this PR and add a comment once I have done a once over.~

The only thing I am seeing that is not working is the SCCA Membership Verification on the registration site. That does not have anything to do with these changes, so I documented it [in the wiki](https://github.com/azbrscca/azbrscca.org/wiki/Registration:-Known-Issues#scca-membership-verification). The short version is, SCCA's service is whitelisted by IP so I need to email them for an update. I'll copy `webmaster@azbrscca.org` on the email.
